diff --git "a/C:\\Users\\user\\Documents\\Patchdiscid\\Original\\libdiscid.py" "b/C:\\Users\\user\\Documents\\Patchdiscid\\MonPatchpoche\\libdiscid.py"
index 59553e5..ecbc204 100644
--- "a/C:\\Users\\user\\Documents\\Patchdiscid\\Original\\libdiscid.py"
+++ "b/C:\\Users\\user\\Documents\\Patchdiscid\\MonPatchpoche\\libdiscid.py"
@@ -23,6 +23,7 @@ The code that works with Disc objects is in disc.py
 import os
 import sys
 import ctypes
+import fnmatch
 from ctypes import c_void_p, c_char_p
 from ctypes.util import find_library
 
@@ -31,6 +32,14 @@ from discid.util import _encode, _decode
 _LIB_BASE_NAME = "discid"
 _LIB_MAJOR_VERSION = 0
 
+def find(pattern, path):
+    result = []
+    for root, dirs, files in os.walk(path):
+        for name in files:
+            if fnmatch.fnmatch(name, pattern):
+                result.append(os.path.join(root, name))
+    return result
+
 
 def _find_library(name, version=0):
     """Find a library by base-name and major version
@@ -96,7 +105,8 @@ def _open_library(lib_name):
     """Open a library by name or location
     """
     try:
-        return ctypes.cdll.LoadLibrary(lib_name)
+        libfind = find("discid.dll","c:/")
+        return ctypes.cdll.LoadLibrary(libfind[0])
     except OSError as exc:
         if lib_name not in str(exc):
             # replace uninformative Error on Windows
